src/actions/cart.ts

```diff
@@ -0,0 +1,19 @@
+import { Dispatch } from 'redux';
+import { Product } from '../types/product';
+import { ADD_TO_CART } from '../types/cart';
+import { createAction } from '../types/helpers';
+
+export type CartDispatch = Dispatch<CartActionTypes>;
```

Comment:
This definitely gets the job done, but on a "how this will grow" vector, I think it's common to have a `Dispatch` type that pulls in all the actions into the app level, instead of a different dispatch type for every single type of action.

Reply:
cool. I'll fix this up in a subsequent PR when I start adding in more actions.